mtags/src/main/scala-2/scala/meta/internal/pc/SemanticTokenProvider.scala

```diff
@@ -0,0 +1,571 @@
+      tpe.original match {
+        case cp.AppliedTypeTree(tpt, _) => tpt.pos
+        case sel: cp.Select => sel.namePos
+        case _ => tpe.pos
+      }
+    }
+    // We need to collect named params since they will not show on fully typed tree
+    lazy val namedArgCache = {
+      val parsedTree = cp.parseTree(source)
+      parsedTree.collect { case arg @ cp.AssignOrNamedArg(_, rhs) =>
+        rhs.pos.start -> arg
+      }.toMap
+    }
+  }
+
+  def selector(imp:cp.Import, startOffset:Int):Option[cp.Symbol] ={
+    for {sel <- imp.selectors.reverseIterator
+      .find(_.namePos <= startOffset)
+    } yield imp.expr.symbol.info.member(sel.name)
+  }
+  def pickFromTraversed(tk:scala.meta.tokens.Token):NodeInfo ={
+      val buffer = ListBuffer.empty[NodeInfo]
+        
+      for (node <- nodes){
+        node.tree match {
+          case imp:cp.Import =>
+            selector(imp,tk.pos.start) match {
+              case Some(sym) =>buffer.addAll(List(NodeInfo(sym)))
+              case None => //pass
+            }
+          case _ =>
+            if( node.pos.start ==tk.pos.start &&
+              node.pos.end ==tk.pos.end 
+              //  node.tree.symbol.name.toString==tk.text
+            ) buffer.addAll(List(node))
+        }
+      }
+
+      val nodeList = buffer.toList
+      if (nodeList.size ==0) null else nodeList(0)
+  }
+
+
+  /**
+    * returns (SemanticTokenType, SemanticTokenModifier) of @param tk
+    */
+  private def getTypeAndMod(tk:scala.meta.tokens.Token):(Int, Int,String) ={
+
+    // whether token is identifier or not
+    tk match {
+      case _: Token.Ident|_: Token.Constant.Symbol => 
+          // continue this method.
+          // Constant.Symbol means literal symbol with backticks.
+          // e.g. which is `yield` of such as Thread.`yield`().
+      case _ =>
+         //Non-Ident has no modifier.
+         return (typeOfNonIdentToken(tk), 0, strSep + "Non-Ident")
+    }
+
+    var logString = ""
+    logString =logString+ linSep + linSep  + "  Start:Ident Part getSemanticTypeAndMod"
+    logString =logString+ linSep + " txt: " + tk.text
+
+    val nodeInfo = pickFromTraversed(tk)
+    
+    if( nodeInfo == null) {(-1,0,strSep + "Node-Nothing")}
+    else {
+      val sym = nodeInfo.symbol
+
+      //Moodifier to return
+      var mod:Int = 0
+      def addPwrToMod(tokenID:String)={
+        val place:Int = capableModifiers.indexOf(tokenID)
+        if (place != -1) mod += scala.math.pow(2, place).toInt
+      }
+
+      //get Type
+      val typ = 
+          if (sym.isValueParameter
+          // |sym.isParamWithDefault
+          // |sym.isDefaultGetter
+          // |sym.isParamAccessor
+          // |sym.isGetter
+          // |sym.isSetter
+          ) getTypeId(SemanticTokenTypes.Parameter)
+          else if (sym.isTypeParameter) getTypeId(SemanticTokenTypes.TypeParameter)
+          else if (sym.isJavaEnum) getTypeId(SemanticTokenTypes.Enum)
```

Comment:
Could you comment it temporary? It's causing problems with Scala 2.11.12 compilation
We can think of solution after making CI green.

Reply:
I deleted the part and mad a note below.